mern-ecommerce/controllers/order.js

```diff
@@ -61,3 +61,17 @@ exports.updateOrderStatus = (req, res) => {
     }
   );
 };
+
+exports.deleteOrderById = (req, res) => {
+  Order.deleteOne({ _id: req.params.orderId })
+    .exec()
+    .then(result => {
+      if (result.deletedCount === 0) {
+        return res.status(404).send({ error: "No order found with that ID" });
+      }
+      res.status(200).send({ message: "Successfully deleted the order" });
+    })
+    .catch(err => {
+      res.status(500).send({ error: errorHandler(err) });
```

Comment:
The error response should include a more descriptive error message. Instead of just sending `{ error: errorHandler(err) }`, consider sending `{ error: errorHandler(err), message: 'Failed to delete order' }`. This will provide more context to the client about what went wrong.